ticketutil/servicenow.py

```diff
@@ -0,0 +1,410 @@
+         'pending vendor':'-5',
+         'resolved':'5',
+         'closed completed':'3',
+         'closed cancelled':'8'}
+
+
+class ServiceNowTicket(Ticket):
+    """
+    ServiceNow Ticket object. Contains ServiceNow specific methods for working
+    with tickets.
+    """
+    def __init__(self, url, project, auth=None, ticket_id=None):
+        """
+        :param url: ServiceNow service url
+        :param project: ServiceNow table or project
+        :param auth: (<username>, <password>) for HTTP Basic Authentication
+        :param ticked_id: ticked number, eg. 'PNT1234567'
+        """
+        self.ticketing_tool = 'ServiceNow'
+
+        # The auth param should be of the form (<username>, <password>) for
+        # HTTP Basic authentication.
+        self.url = url
+        self.auth = auth
+        self.table = project
+        self.rest_url = '{0}/api/now/v1/table/{1}'.format(self.url, self.table)
+        self.auth_url = self.rest_url
+        self.headers_post_ = {'Content-Type': 'application/json',
+                              'Accept': 'application/json'}
+        # Call our parent class's init method which creates our requests session.
+        super(ServiceNowTicket, self).__init__(project, ticket_id)
+
+    def _verify_project(self, project):
+        """
+        Queries the ServiceNow API to see if project is a valid table for
+        the given ServiceNow instance.
+        :param project: The project table you're verifying.
+        :return: True or False depending on if project is valid.
+        """
+        try:
+            r = self.s.get('{0}?sysparm_limit=1'.format(self.rest_url))
+            logging.debug("Verify project: Status Code: {0}".format(r.status_code))
+            r.raise_for_status()
+            logging.debug("Project {0} is valid".format(project))
+            return True
+        except request.RequestException as e:
+            logging.error("Unexpected error occurred when verifying project.")
+            logging.error(e.args)
+            return False
+
+    def get_ticket_content(self, ticket_id=None):
+        """
+        Get ticket_content using ticket_id
+
+        :param ticked_id: ticked number, if not set self.ticket_id is used
+        :return: ticket content
+        """
+        if ticket_id is None:
+            ticket_id = self.ticket_id
+        try:
+            self.s.headers.update({'Content-Type': 'application/json'})
+            url = self.rest_url + '?sysparm_query=GOTOnumber%3D'
+            url += ticket_id
+            r = self.s.get(url)
+            r.raise_for_status()
+
+            logging.debug("Get ticket content: Status Code: {0}"
+                          .format(r.status_code))
+            ticket_content = r.json()
+            return ticket_content['result'][0]
+        except requests.RequestException as e:
+            logging.error("Error while getting ticket content")
+            logging.error(e.args)
+            return False
```

Comment:
Is this `return False` designed as an error handling in the whole project? I would recommend have this consistent, as this does not seems to be the case, see:
```
grep "except requests.RequestException as e:" -A 3 ticketutil/*.py
```
I would recommend to reraise the error by `raise` instead of returning `False` and let the user decide what to do in case of error... @dmranck, what is your opinion on this?